### PR TITLE
Gradle Plugin Sample taking advantage of `kotlin-dsl`

### PR DESCRIPTION
**Pull request by eskatos**

### CODE REVIEW

File: samples/gradle-plugin/build.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,14 @@
+tasks {
+
+    val publishPlugin by creating(GradleBuild::class) {
+        dir = file("plugin")
+        tasks = listOf("publish")
+    }
+
+    val checkSample by creating(GradleBuild::class) {
+        dir = file("sample")
```
Comment (bamboo) [inline samples/gradle-plugin/build.gradle.kts]: ❌ I think `consumer` would be a better name. What do you think?

File: samples/gradle-plugin/build.gradle.kts [hunk 2]
```diff
@@ -0,0 +1,14 @@
+tasks {
+
+    val publishPlugin by creating(GradleBuild::class) {
+        dir = file("plugin")
+        tasks = listOf("publish")
+    }
+
+    val checkSample by creating(GradleBuild::class) {
```
Comment (bamboo) [inline samples/gradle-plugin/build.gradle.kts]: :x: The task in the sample should make sense on its own. I think `consumePlugin` would be a better name, see below.

**Comment by eskatos** [comment]
Thanks @bamboo for the feedback.
Comments addressed, could you have another look?

File: samples/gradle-plugin/build.gradle.kts [hunk 3]
```diff
@@ -0,0 +1,14 @@
+tasks {
+
+    val publishPlugin by creating(GradleBuild::class) {
+        dir = file("plugin")
+        tasks = listOf("publish")
+    }
+
+    val checkConsumer by creating(GradleBuild::class) {
```
Comment (bamboo) [inline samples/gradle-plugin/build.gradle.kts]: I'm a bit bothered by the `check` prefix here.

How about renaming `publishPlugin` to just `plugin` and `checkConsumer` to just `consumer`?

Comment (eskatos) [inline samples/gradle-plugin/build.gradle.kts]: Sounds good.
Done

**Pull request merged**